Client/Frontend/Browser/Playlist/VideoPlayer/VideoPlayer.swift

```diff
@@ -0,0 +1,698 @@
+        self.playObserver = self.player.addPeriodicTimeObserver(forInterval: interval, queue: .main, using: { [weak self] time in
+            guard let self = self, let currentItem = self.player.currentItem else { return }
+            
+            let endTime = CMTimeConvertScale(currentItem.asset.duration, timescale: self.player.currentTime().timescale, method: .roundHalfAwayFromZero)
+            
+            if CMTimeCompare(endTime, .zero) != 0 && endTime.value > 0 {
+                self.controlsView.trackBar.setTimeRange(currentTime: self.player.currentTime(), endTime: endTime)
+            }
+        })
+    }
+    
+    private func registerPictureInPictureNotifications() {
+        if AVPictureInPictureController.isPictureInPictureSupported() {
+            pictureInPictureController = AVPictureInPictureController(playerLayer: self.playerLayer)
+            guard let pictureInPictureController = pictureInPictureController else { return }
+            
+            pictureInPictureObservers.append(pictureInPictureController.observe(\AVPictureInPictureController.isPictureInPicturePossible, options: [.initial, .new]) { [weak self] _, change in
+                self?.infoView.pictureInPictureButton.isEnabled = change.newValue ?? false
+            })
+        } else {
+            infoView.pictureInPictureButton.isEnabled = false
+        }
+    }
+    
+    private func showOverlays(_ show: Bool) {
+        self.showOverlays(show, except: [], display: [])
+    }
+    
+    private func showOverlays(_ show: Bool, except: [UIView] = [], display: [UIView] = []) {
+        var except = except
+        var display = display
+        
+        if !isVideoAvailable() {
+            // If the overlay is showing, hide the particle view.. else show it..
+            except.append(particleView)
+            
+            if !show {
+                display.append(particleView)
+            }
+        } else {
+            if show {
+                except.append(particleView)
+            }
+        }
+        
+        UIView.animate(withDuration: 1.0, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 1.0, options: [.curveEaseInOut, .allowUserInteraction], animations: {
+            self.subviews.forEach({
+                if !except.contains($0) {
+                    $0.alpha = show ? 1.0 : 0.0
+                } else if display.contains($0) {
+                    $0.alpha = 1.0
+                } else {
+                    $0.alpha = 0.0
+                }
+            })
+        })
+    }
+    
+    public func setVideoInfo(videoDomain: String, videoTitle: String?) {
+        var displayTitle = videoTitle ?? ""
+        
+        if displayTitle.isEmpty {
+            var hostDomain = ""
+            
+            if let host = URL(string: videoDomain)?.host {
+                hostDomain = host
+                if host.hasPrefix("www.") {
+                    hostDomain = String(host.dropFirst(4))
```

Comment:
don't we have a URL helper method for stripping domain from `www./m.` subdomains?